source/common/config/utility.cc

```diff
@@ -119,5 +123,13 @@ std::string Utility::resourceName(const ProtobufWkt::Any& resource) {
       fmt::format("Unknown type URL {} in DiscoveryResponse", resource.type_url()));
 }
 
+void Utility::checkUserSuppliedNameLength(const std::string& error_prefix,
+                                          const std::string& name) {
+  if (name.length() > Stats::RawStatData::maxUserSuppliedNameLength()) {
+    throw EnvoyException(fmt::format("{}: name exceeds allowed maximum {}", error_prefix,
+                                     Stats::RawStatData::maxUserSuppliedNameLength()));
```

Comment:
Please also print the passed in name, and possibly the length, so it's easier to figure out what was too big and by how much it was too big.

Reply:
I can, but be warned that its of little use. The fields are named "name" which is not of much use. Thats the reason I opted to use the error_prefix and have some meaningful error message as part of the error.